src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
```

Comment:
suggestion of implementation with reduce method:
<img width="554" alt="image" src="https://github.com/mate-academy/ts_group_by_key/assets/87986955/ea7de28f-6116-4b0b-8a6e-7bf6616ce92d">